ftw/publisher/sender/FormGen/saveDataAdapter.py

```diff
@@ -0,0 +1,50 @@
+from ftw.publisher.sender import _
+from ftw.publisher.sender.utils import sendRequestToRealm
+from ftw.publisher.sender.interfaces import IConfig
+from ftw.publisher.sender.workflows.interfaces import IPublisherContextState
+from Products.statusmessages.interfaces import IStatusMessage
+from Products.CMFCore.utils import getToolByName
+from zope.component import getUtility
+from Acquisition import aq_parent
+from Products.Archetypes.utils import contentDispositionHeader
+from zope.component import getMultiAdapter
+
+
+def download(self, REQUEST=None, RESPONSE=None):
+    """Download the saved data
+    """
+    url_tool = getToolByName(self, 'portal_url')
+    config = IConfig(url_tool.getPortalObject())
+    pub_state = getMultiAdapter((self, REQUEST), IPublisherContextState)
+    realms = config.getRealms()
+    format = getattr(self, 'DownloadFormat', 'csv')
+
+    if len(realms) == 0 or not pub_state.is_parent_published():
```

Comment:
Fair enough.
Can we add a status message "data downloaded from backend" / "... frontend"? 
Or is that impossible because of redirect / download combination?
_Should_ we do that?